File: 0888/888.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var fairCandySwap = function(root) {
    if (!root) return 0;
    return 1 + Math.max(fairCandySwap(root.left), fairCandySwap(root.right));
};

File: 0888/888.go
func fairCandySwap(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := fairCandySwap(root.Left)
	r := fairCandySwap(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}